packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -186,16 +186,126 @@ class CupertinoThemeData with Diagnosticable {
   );
 
   const CupertinoThemeData._rawWithDefaults(
-    this.brightness,
-    this._primaryColor,
-    this._primaryContrastingColor,
-    this._textTheme,
-    this._barBackgroundColor,
-    this._scaffoldBackgroundColor,
+    Brightness? brightness,
+    Color? primaryColor,
+    Color? primaryContrastingColor,
+    CupertinoTextThemeData? textTheme,
+    Color? barBackgroundColor,
+    Color? scaffoldBackgroundColor,
     this._defaults,
+  ) : super(
+    brightness: brightness,
+    primaryColor: primaryColor,
+    primaryContrastingColor: primaryContrastingColor,
+    textTheme: textTheme,
+    barBackgroundColor: barBackgroundColor,
+    scaffoldBackgroundColor: scaffoldBackgroundColor,
   );
 
-  final _CupertinoThemeDefaults? _defaults;
+  final _CupertinoThemeDefaults _defaults;
+
+  @override
+  Color get primaryColor => super.primaryColor ?? _defaults.primaryColor;
+
+  @override
+  Color get primaryContrastingColor => super.primaryContrastingColor ?? _defaults.primaryContrastingColor;
+
+  @override
+  CupertinoTextThemeData get textTheme {
+    return super.textTheme ?? _defaults.textThemeDefaults.createDefaults(primaryColor: primaryColor);
+  }
+
+  @override
+  Color get barBackgroundColor => super.barBackgroundColor ?? _defaults.barBackgroundColor;
+
+  @override
+  Color get scaffoldBackgroundColor => super.scaffoldBackgroundColor ?? _defaults.scaffoldBackgroundColor;
+
+  @override
+  NoDefaultCupertinoThemeData noDefault() {
+    return NoDefaultCupertinoThemeData(
+      brightness: super.brightness,
+      primaryColor: super.primaryColor,
+      primaryContrastingColor: super.primaryContrastingColor,
+      textTheme: super.textTheme,
+      barBackgroundColor: super.barBackgroundColor,
+      scaffoldBackgroundColor: super.scaffoldBackgroundColor,
+    );
+  }
+
+  @protected
```

Comment:
Shouldn't this `protected` be on `NoDefaultCupertinoThemeData`?

Reply:
Totally. Fixed.